# 1. Calculadora de integrales en python

### Construccion de la funcion

In [1]:
import numpy as np

# Función que aproxima una integral definida usando Monte Carlo
def integral_montecarlo(f, a, b, N=10000):
    
    # 1. Generar N muestras uniformes en [a,b]
    x = np.random.uniform(a, b, N)
    
    # 2. Evaluar la función en cada muestra
    fx = f(x)
    
    # 3. Calcular el promedio y multiplicar por (b-a)
    integral = (b - a) * np.mean(fx)
    
    return integral


### Prueba con la función $f(x) = x^2$

$$
\int_{0}^{1} x^2 \, dx
$$

El valor exacto de esta integral es:

$$
\int_{0}^{1} x^2 \, dx = \left[ \frac{x^3}{3} \right]_{0}^{1} = \frac{1}{3} \approx 0.3333
$$

Usaremos el método de Monte Carlo para estimar este valor y compararlo con la solución exacta.

In [2]:
# Definimos la función f(x) = x^2
def f1(x):
    return x**2

# Parámetros
a = 0
b = 1
N = 10000

# Aproximación Monte Carlo
aprox = integral_montecarlo(f1, a, b, N)

# Valor exacto
valor_exacto = 1/3

print("Aproximación Monte Carlo:", aprox)
print("Valor exacto:", valor_exacto)
print("Error:", abs(aprox - valor_exacto))


Aproximación Monte Carlo: 0.3341542559195689
Valor exacto: 0.3333333333333333
Error: 0.0008209225862355618


Al probar la función f(x)=x² en el intervalo [0,1], se obtuvo una aproximación de 0.33415, mientras que el valor exacto es 0.33333. El error fue de aproximadamente 0.00082, lo cual muestra que el método de Monte Carlo proporciona una estimación bastante cercana incluso con 10,000 simulaciones. Se comprobó que el error disminuye al aumentar el número de muestras.
